src/serac/physics/utilities/finite_element_dual.hpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file finite_element_dual.hpp
+ *
+ * @brief This contains a class that represents the space of topological duals sampled
+ * on a discrete finite element basis, i.e. the space of residuals and sensitivities.
+ */
+
+#pragma once
+
+#include "mfem.hpp"
+
+#include "serac/physics/utilities/finite_element_state.hpp"
+
+namespace serac {
+
+/**
+ * @brief Class for encapsulating the topological dual space of a finite element space (i.e. the
+ * space of linear forms) as applied to a specific basis set
+ *
```

Comment:
There should be some examples here to help the reader.

And I think you want to get rid of the word "topological", it's confusing to the reader.  It's just the dual space to the finite element space, e.g. the space of reactions.

Reply:
The distinction between algebraic and topological dual spaces is only material in the infinite-dimensional case, and for an engineering audience, I can see the appeal in eliding that distinction.